lib/astutils.cpp

```diff
@@ -2188,15 +2188,15 @@ T* getTokenArgumentFunctionImpl(T* tok, int& argn)
     argn = -1;
     {
         T* parent = tok->astParent();
-        if (parent && parent->isUnaryOp("&"))
+        if (parent && (parent->isUnaryOp("&") || parent->isIncDecOp()))
             parent = parent->astParent();
         while (parent && parent->isCast())
             parent = parent->astParent();
         if (Token::Match(parent, "[+-]") && parent->valueType() && parent->valueType()->pointer)
             parent = parent->astParent();
 
         // passing variable to subfunction?
-        if (Token::Match(parent, "[[(,{]"))
+        if (Token::Match(parent, "[*[(,{]"))
```

Comment:
`indirect` should decremented if `*` is used.

Reply:
That's `getTokenArgumentFunctionImpl()`, so there is no `indirect`. It seems that `getTokenArgumentFunction()` just returned `nullptr` when there was a `*`.

Reply:
Yes, but we need to decrement `indirect` still in the `isVariableChangedByFunctionCall` like how we increment it when it finds a `&` operator.